distsql: move physical plan basics to distsqlplan

RaduBerinde opened the pull request:
Split off the basic primitives of the planning code - processsors, streams, stages,
projections, renders, limits - into distsqlplan.

No code changes, except some minor cleanup in a couple of instances.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13511)
<!-- Reviewable:end -->

RaduBerinde committed:
distsql: add a few TestDistSQLPlanner cases

Adding two more cases to `TestDistSQLPlanner/Basic`: the existing cases were
using a single table reader because of the constraint on `y`.

RaduBerinde commented:
@andreimatei Please take a look at this today if you can. It shouldn't take more than a couple of minutes since it's just moving code around.

andreimatei commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/sql/distsql_physical_planner.go, line 322 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13511#-KcxM5Lrh1H29ouJql00:-KcxM5Lrh1H29ouJql01:bcbli7w) ([raw file](https://github.com/cockroachdb/cockroach/blob/a1fe9e735f4e27dd0cc2b04a224bc485e5b1036f/pkg/sql/distsql_physical_planner.go#L322)):*
> ```Go
> //
> // It augments distsqlplan.PhysicalPlan with information relating the physical
> // plant to the planNode.
> ```

is "plant to the planNode"  what you wanted to say?

---

*[pkg/sql/distsql_physical_planner_test.go, line 132 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13511#-KcxNLLSbSzZgAnzvPg3:-KcxNFbkLXqrjyfQFgIz:bagypga) ([raw file](https://github.com/cockroachdb/cockroach/blob/8d2113ae7444a0598418e9e4f3cae2b1b9683023/pkg/sql/distsql_physical_planner_test.go#L132)):*
> ```Go
> 		)
> 		r.CheckQueryResults(
> 			"SELECT 5, 2 + y, * FROM NumToStr WHERE y % 1000 = 0 ORDER BY str",
> ```

mind adding comments to these queries to explain how the indexes used /planning differ? Or make them different subtests.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13511#-:-KcxNUYa8rU826eYzq-C:bnfp4nl)*
<!-- Sent from Reviewable.io -->

RaduBerinde committed:
distsql: move physical plan basics to distsqlplan

Split off the basic primitives of the planning code - processsors, streams,
stages, projections, renders, limits - into distsqlplan.

RaduBerinde committed:
distsqlrun: implement diagram block for backfiller

RaduBerinde commented:
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/distsql_physical_planner.go, line 322 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13511#-KcxM5Lrh1H29ouJql00:-KcxWqIAlUzRu5QZUbm1:bt2vr1b) ([raw file](https://github.com/cockroachdb/cockroach/blob/a1fe9e735f4e27dd0cc2b04a224bc485e5b1036f/pkg/sql/distsql_physical_planner.go#L322)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

is "plant to the planNode"  what you wanted to say?
</blockquote></details>

Fixed.

---

*[pkg/sql/distsql_physical_planner_test.go, line 132 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/13511#-KcxNLLSbSzZgAnzvPg3:-KcxWr9-_hyxGiLphkaz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8d2113ae7444a0598418e9e4f3cae2b1b9683023/pkg/sql/distsql_physical_planner_test.go#L132)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

mind adding comments to these queries to explain how the indexes used /planning differ? Or make them different subtests.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13511)*
<!-- Sent from Reviewable.io -->
